core/extension/gdextension_manager.cpp

```diff
@@ -114,7 +115,14 @@ GDExtensionManager::LoadStatus GDExtensionManager::load_extension(const String &
 
 	Ref<GDExtensionLibraryLoader> loader;
 	loader.instantiate();
-	return GDExtensionManager::get_singleton()->load_extension_with_loader(p_path, loader);
+	return load_extension_with_loader(p_path, loader);
+}
+
+GDExtensionManager::LoadStatus GDExtensionManager::load_function_extension(const String &p_path, GDExtensionConstPtr<const GDExtensionInitializationFunction> p_init_func) {
```

Comment:
I gave some review on this function name [here](https://github.com/godotengine/godot/pull/90510#discussion_r1755763934) in the predecessor PR:

> Another naming nitpick: "function extension" is an unclear term that we don't really use anywhere else.
> 
> Maybe `load_extension_from_function()`?
> 
> Or, if we can decide on an overall name for this feature we could use that term. So, if we standardize on "LibGodot", then this could be `load_libgodot_extension()`? Or, if we standardize on "embedded", then it could be `load_embedded_extension()`?

Reply:
Yeah, this is a bit of a weird one. For reference, the point of this is to create something "extension-like" that is simply based on a callback, rather than loading an entire dynamic library (which we can't do because, in this environment, Godot *is* the library.) This feature is a bit overbuilt for the purposes of libgodot but I figured I'd leave it in because other people seem to want it.

I'd personally lean towards "load_extension_from_function()", but I'm not wedded to any of the terminology here and will happily change on request.

Reply:
> I'd personally lean towards "load_extension_from_function()", but I'm not wedded to any of the terminology here and will happily change on request.

Please, go ahead! That function name is better because it's description of what it's doing, rather than inventing the new term "function extension"